Unicode/ustring.c

```diff
@@ -576,6 +576,10 @@ char *u2utf8_strcpy(char *utf8buf,const unichar_t *ubuf) {
     return( NULL );
 }
 
+char *u2utf8_strcpy(char *utf8buf,const unichar_t *ubuf) {
+    return u2utf8_strncpy(utf8buf, ubuf, u_strlen(ubuf));
```

Comment:
Now we're looping twice for `u2utf8_strcpy()`, once to get the length to pass in and once to copy the chars. Almost every code review comment is about tidying up; time to feel dirty. 

Take a deep breath, in/out.

Now: consider passing -1 as the length here.